step-plans/step-plans-core/src/main/java/step/core/execution/ExecutionEngineRunner.java

```diff
@@ -229,6 +233,12 @@ private ReportNode execute(Plan plan, ReportNode rootReportNode) throws Provisio
 			updateStatus(ExecutionStatus.RUNNING);
 			return artefactHandlerManager.execute(root, rootReportNode);
 		} finally {
+			try {
+				//Flush report node TS
+				executionContext.require(ReportNodeTimeSeries.class).flush();

```

Comment:
This should be done per hook in the related plugin. For instance with the hook step.core.plugins.ExecutionCallbacks#deprovisionRequiredResources